bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericMetadataProvider.java

```diff
@@ -86,7 +119,7 @@ public void removeMetadata(String itemName) {
     public Collection<Metadata> getAll() {
         try {
             lock.readLock().lock();
-            return Collections.unmodifiableSet(metadata);
+            return Collections.unmodifiableCollection(metadata.values());
```

Comment:
Shouldn't this be changed to

```Java
return Collections.unmodifiableCollection(new HashSet<>(metadata.values()));
```

`values()` of `HashMap`:

> Returns a Collection view of the values contained in this map. The collection is backed by the map, so changes to the map are reflected in the collection, and vice-versa. If the map is modified while an iteration over the collection is in progress (except through the iterator's own remove operation), the results of the iteration are undefined. The collection supports element removal, which removes the corresponding mapping from the map, via the Iterator.remove, Collection.remove, removeAll, retainAll and clear operations. It does not support the add or addAll operations.

We return an unmodifiable **view** for a **view** of the map value's.

We still potentially break the caller if we modify our internally hold map.

Reply:
Yes, correct. Better save than sorry. Do you mind creating a change?

Reply:
Yes, I put it on my todo list.

Reply:
Obsolete by 683dccd33701b6a484bbff7316f206f11d1fd9d0 integrated in #5873 :wink: